core/.js/src/main/scala/io/udash/utils/FileService.scala

```diff
@@ -0,0 +1,45 @@
+package io.udash.utils
+
+import java.io.IOException
+
+import org.scalajs.dom._
+import scalatags.JsDom.all._
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.typedarray._
+
+import scala.concurrent._
+import scala.util.Try
+
+object FileService {
+  def save(filename: String, bytes: Array[Byte], mimeType: String = "application/octet-stream"): Unit = {
+    val jsBytes = Array[js.Any](bytes.toTypedArray).toJSArray
+    val blob = new Blob(jsBytes, BlobPropertyBag(mimeType))
+    val url = URL.createObjectURL(blob)
+    val download = attr("download")
+
+    val anchor = a(href := url, download := filename, style := "display: none")(filename).render
+    document.body.appendChild(anchor)
+    anchor.click()
```

Comment:
I understand that this is your use case, but I think even more common one could be to render the anchor in the DOM. We should either provide less functionality or more granular one in library code.

Reply:
Very good point, thanks.